core/src/main/java/org/jdbi/v3/core/mapper/BuiltInMapperFactory.java

```diff
@@ -87,6 +87,8 @@
         mappers.put(OffsetDateTime.class, referenceMapper(BuiltInMapperFactory::getOffsetDateTime));
         mappers.put(ZonedDateTime.class, referenceMapper(BuiltInMapperFactory::getZonedDateTime));
         mappers.put(LocalTime.class, referenceMapper(BuiltInMapperFactory::getLocalTime));
+
+        mappers.put(Object.class, referenceMapper(ResultSet::getObject));
```

Comment:
I found an alternative approach that preserves the tested behavior without having a system-wide Object mapper.